apis/v1alpha1/httproute_types.go

```diff
@@ -337,7 +337,7 @@ type HTTPRouteMatch struct {
 	//
 	// +optional
 	// +kubebuilder:default={type: "Prefix", value: "/"}
-	Path HTTPPathMatch `json:"path,omitempty"`
+	Path *HTTPPathMatch `json:"path,omitempty"`
```

Comment:
Interesting. `HTTPHeaderMatch` is a pointer because there is no default value for it. This was not a pointer because this couldn't have a `nil` value since it has a default.